uk_vat_service.go

```diff
@@ -0,0 +1,49 @@
+package vat
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+)
+
+// ukVATService is service that calls a UK VAT API to validate UK VAT numbers.
+type ukVATService struct{}
+
+// Validate checks if the given VAT number exists and is active. If no error is returned, then it is.
+func (s *ukVATService) Validate(vatNumber string) error {
+
+	prefix := vatNumber[0:2]
+	number := vatNumber[2:]
+
+	// Only VAT numbers starting with "GB" are supported by this service. All others should go through the VIES service.
+	if prefix != "GB" {
+		return ErrInvalidCountryCode
+	}
```

Comment:
This has a chance to panic due to slice out of bounds range, if the string passed is shorter than 2 characters 😅 Also people don't always type stuff in uppercase (you could argue we shouldn't allow lowercase, but still)...

You can probably do something like `if !strings.HasPrefix(strings.ToUpper(vatNumber), "GB")` check for country code first, then do a `len(vatNumber) > 2` check (error if too short) and then finally grab the number part (`2:` or however) 😊

Reply:
So the code that calls this actually has this check:

```
if strings.HasPrefix(vatNumber, "GB") {
lookupService = UKVATLookupService
}
```

So it should always have that prefix, but I guess this service has no reason to know the code that called it did that, so we can run the check again in here.